src/crepe/system/CollisionSystem.cpp

```diff
@@ -1,5 +1,393 @@
+
+	// Check if there is a collision should always be true
+	if (overlap_x > 0 && overlap_y > 0) {
+		// Determine the direction of resolution
+		if (overlap_x < overlap_y) {
+			// Resolve along the X-axis (smallest overlap)
+			resolution.x = (delta.x > 0) ? -overlap_x : overlap_x;
+		} else if (overlap_y < overlap_x) {
+			// Resolve along the Y-axis (smallest overlap)
+			resolution.y = (delta.y > 0) ? -overlap_y : overlap_y;
+		} else {
+			// Equal overlap, resolve both directions with preference
+			resolution.x = (delta.x > 0) ? -overlap_x : overlap_x;
+			resolution.y = (delta.y > 0) ? -overlap_y : overlap_y;
+		}
+	}
+
+	return resolution;
+}
+
+void CollisionSystem::determine_collision_handler(CollisionInfo& info){
+	// Check rigidbody type for static
+	if(info.first_rigidbody.data.body_type != Rigidbody::BodyType::STATIC)
+	{
+		// If second body is static perform the static collision handler in this system
+		if(info.second_rigidbody.data.body_type == Rigidbody::BodyType::STATIC){
+			static_collision_handler(info);
+		}; 
+		// Call collision event for user
+		CollisionEvent data(info);
+		EventManager::get_instance().trigger_event<CollisionEvent>(data, info.first_collider.game_object_id);
+	}		
+}
+
+void CollisionSystem::static_collision_handler(CollisionInfo& info){
+	// Move object back using calculate move back value 
+	info.first_transform.position += info.resolution;
+
+	// If bounce is enabled mirror velocity
+	if(info.first_rigidbody.data.bounce) {
+		if(info.resolution_direction == Direction::BOTH)
+		{
+			info.first_rigidbody.data.linear_velocity.y = -info.first_rigidbody.data.linear_velocity.y * info.first_rigidbody.data.elastisity;
+			info.first_rigidbody.data.linear_velocity.x = -info.first_rigidbody.data.linear_velocity.x * info.first_rigidbody.data.elastisity;
+		}
+		else if(info.resolution_direction == Direction::Y_DIRECTION) {
+			info.first_rigidbody.data.linear_velocity.y = -info.first_rigidbody.data.linear_velocity.y * info.first_rigidbody.data.elastisity;
+		}
+		else if(info.resolution_direction == Direction::X_DIRECTION){
+			info.first_rigidbody.data.linear_velocity.x = -info.first_rigidbody.data.linear_velocity.x * info.first_rigidbody.data.elastisity;
+		}
+	}
+	// Stop movement if bounce is disabled
+	else {
+		info.first_rigidbody.data.linear_velocity = {0,0};
+	}
+}
+
+std::vector<std::pair<CollisionSystem::CollisionInternal,CollisionSystem::CollisionInternal>> CollisionSystem::check_collisions(std::vector<collider_variant> & colliders) {
+	
+	
+	// TODO:
+	// If no colliders skip
+	// Check if colliders has rigidbody if not skip
+
+	// TODO:
+	// If amount is higer than lets say 16 for now use quadtree otwerwise skip
+	// Quadtree code
+	// Quadtree is placed over the input vector
+
+	std::vector<std::pair<CollisionInternal,CollisionInternal>> collisions_ret;
+	for (size_t i = 0; i < colliders.size(); ++i) {
+    std::visit([&](auto& inner_collider_ref) {
```

Comment:
`auto&` is hier eigenlijk niet toegestaan volgens de code standaard, maar als je hier gewoon `CollisionInternal&` neerzet hoef je ook meteen geen `.get()` meer aan te roepen, en krijg je fatsoenlijke completion in vscode.